### PR TITLE
Halloween update

### PR DESCRIPTION
**Pull request by Woafflation**
1. Added store to sync all halloween components
2. Added Academy banner
3. Added Pricing page banner
4. Added Pumpkin component
5. Added halloween popup and trigger button

https://www.notion.so/santiment/Halloween-promo-campaign-22-19753e6111784456865888f17daf7f77

Снимок экрана 2022-10-27 в 16 31 02
Снимок экрана 2022-10-27 в 16 31 06
Снимок экрана 2022-10-27 в 16 31 11
Снимок экрана 2022-10-27 в 16 31 15
Снимок экрана 2022-10-27 в 16 31 24
Снимок экрана 2022-10-27 в 16 31 39

### CODE REVIEW

File: src/ui/Product.svelte [hunk 1]
```diff
@@ -1,21 +1,45 @@
...
 </script>
 
-<a href="/" class="body-2 txt-m title row v-center {className}" on:click={window.__onLinkClick}>
-  <img src="{process.env.MEDIA_PATH}/illus/san-logo.svg" alt="SAN" class="mrg-s mrg--r" />
-  {title}
-</a>
+<div class="relative">
+  {#if isHalloween}
+    <Svg illus id="batman" w="57" h="52" class="$style.batman" />
+  {/if}
```
Comment (DmitriVanGuard) [inline src/ui/Product.svelte:13]: `div` is redundant. Render inside `a`

File: src/ui/Product.svelte [hunk 2]
```diff
@@ -1,21 +1,45 @@
 <script lang="ts">
+  import Svg from '@/ui/Svg'
+
   let className = ''
   export { className as class }
   export let title: string
+  export let isHalloween: string
```
Comment (DmitriVanGuard) [inline src/ui/Product.svelte:7]: Should be `boolean`

File: src/ui/Product.svelte [hunk 3]
```diff
@@ -1,21 +1,45 @@
...
+</div>
 
 <style>
   a {
+    z-index: 3;
     color: var(--black);
   }
 
-  img {
+  .product {
```
Comment (DmitriVanGuard) [inline src/ui/Product.svelte:34]: Leave `img` since it's implied that it has to be only `img` element

**Review by DmitriVanGuard — CHANGES_REQUESTED** [review]

File: src/ui/Product.svelte (hunk 1 above)
Comment (Woafflation) [reply src/ui/Product.svelte]: It has to be due to positioning bat behind text and logo

**Review by DmitriVanGuard — CHANGES_REQUESTED** [review]

File: src/ui/Halloween/Pumpkin.svelte [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  function onPumpkinClick() {
+    updatePumpkins(showOn).then(showHalloweenPopup)
+  }
+</script>
+
+{#if isLoggedIn && !pages.includes(showOn)}
+  <button class="btn row hv-center relative {className}" on:click={onPumpkinClick}>
+    <span class="pumpkin">🎃</span>
+    <img src="{process.env.MEDIA_PATH}/illus/halloween/flame.svg" alt="Flame" class="flame" />
+  </button>
```
Comment (DmitriVanGuard) [inline src/ui/Halloween/Pumpkin.svelte:24]: Put inside `FlamePumpkin` component that will receive `className` props and `on:click` handler

File: src/ui/Halloween/HalloweenAcademy.svelte [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+      class="discount"
+    />
+    <p class="body-1 txt-m mrg-s mrg--b">Congratulations!</p>
+    <p class="mrg-m mrg--b">
+      Let’s put your outstanding skills to use and seek an alpha! This promo code is available
+      between now and November 5th!
+    </p>
+    <button class="btn fluid copy expl-tooltip" aria-label={copyLabel} on:click={onCopy}>
+      {discount}
+    </button>
```
Comment (DmitriVanGuard) [inline src/ui/Halloween/HalloweenAcademy.svelte:37]: Put inside `DiscountCode` component

File: src/ui/Halloween/HalloweenAcademy.svelte [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+  onMount(() => pages.length === 3 && mutateCreatePumpkinCode().then((code) => (discount = code)))
+</script>
+
+{#if isLoggedIn && pages.length === 3}
+  <div class="wrapper column hv-center txt-center">
+    <img
+      src="{process.env.MEDIA_PATH}/illus/halloween/halloween-discount-54.svg"
+      alt="Discount"
+      class="discount"
+    />
```
Comment (DmitriVanGuard) [inline src/ui/Halloween/HalloweenAcademy.svelte:29]: Use `<Svg illus id='halloween/halloween-discount-54' w= .... />`

File: src/ui/Halloween/HalloweenPopup.svelte [hunk 7]
```diff
@@ -0,0 +1,203 @@
...
+      <a
+        href="https://app.santiment.net/pricing"
+        class="btn-1 btn--orange body-2"
+        on:click={window.__onLinkClick}
+      >
+        {buttonText}
+      </a>
+      <button class="btn c-waterloo keep mrg-m mrg--t" on:click={() => (isClueShown = true)}>
+        Keep hunting to double down and get 54% discount
+      </button>
```
Comment (DmitriVanGuard) [inline src/ui/Halloween/HalloweenPopup.svelte:107]: And content inside `{#if isClueShown}` should be placed inside `Clue` component

File: src/ui/Halloween/HalloweenBanner.svelte [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+    font-size: 60px;
+    line-height: 73px;
+    margin-bottom: 14px;
+  }
+
+  p {
+    max-width: 425px;
+    line-height: 22px;
+
+    &:last-of-type {
```
Comment (DmitriVanGuard) [inline src/ui/Halloween/HalloweenBanner.svelte:69]: `.rules`

File: src/ui/Halloween/HalloweenBanner.svelte [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+    right: 36px;
+    width: 670px;
+    height: 320px;
+  }
+
+  .sales {
+    width: 181px;
+    height: 18px;
+    margin-top: 24px;
+    margin-bottom: 6px;
```
Comment (DmitriVanGuard) [inline src/ui/Halloween/HalloweenBanner.svelte:56]: margin: 28px 0 6px

**Review by DmitriVanGuard — CHANGES_REQUESTED** [review]

File: src/stores/halloween.ts [hunk 10]
```diff
@@ -0,0 +1,21 @@
+import { query } from '@/api'
+import { QueryStore } from '@/stores/utils'
+
+export const PUMPKINS_COUNT_QUERY = `{
+  getPumpkins
+}`
+
+type Query = SAN.API.Query<'getPumpkins', string[]>
+
+const accessor = ({ getPumpkins }: Query) => new Set([...getPumpkins])
```
Comment (DmitriVanGuard) [inline src/stores/halloween.ts:10]:
```suggestion
const accessor = ({ getPumpkins }: Query) => new Set(getPumpkins)
```

File: src/ui/Halloween/DiscountCode.svelte [hunk 11]
```diff
@@ -0,0 +1,24 @@
+<script lang="ts">
+  import { copy } from '@/utils'
+
+  let className = ''
+  export { className as class }
+  let copyLabel = 'Copy discount'
+  export let discount
```
Comment (DmitriVanGuard) [inline src/ui/Halloween/DiscountCode.svelte:7]:
```suggestion
  export let discount
  
  let copyLabel = 'Copy discount'
```

File: src/ui/Halloween/Dialog.svelte [hunk 12]
```diff
@@ -0,0 +1,127 @@
...
+
+  .btn-1 {
+    width: fit-content;
+    padding: 8px 28px;
+  }
+
+  .copy {
+    padding: 8px 69px;
+    margin-top: -8px;
+    margin-bottom: 24px;
```
Comment (DmitriVanGuard) [inline src/ui/Halloween/Dialog.svelte:125]:
```suggestion
    margin: -8px 0 24px;
```

File: src/ui/Halloween/Dialog.svelte [hunk 13]
```diff
@@ -0,0 +1,127 @@
...
+      pages.size > 1 ? 's' : ''
+    }, and there are ${remainingAmount.toLowerCase()} left. Collect three and receive your 27% off discount off all Santiment products!`
+    buttonText = 'Continue!'
+    padding = 30
+  }
+</script>
+
+<Dialog {...$$props} noTitle bind:closeDialog class="$style.dialog">
+  <button class="btn close" on:click={closeDialog}>
+    <Svg id="close" w="16" h="16" />
```
Comment (DmitriVanGuard) [inline src/ui/Halloween/Dialog.svelte:53]:
```suggestion
    <Svg id="close" w="16" />
```

**Review by DmitriVanGuard — APPROVED** [review]

**Pull request merged**